pkg/syncthing/syncthing.go

```diff
@@ -486,68 +507,108 @@ func (s *Syncthing) GetCompletion(ctx context.Context, local bool, device string
+	if isHealthyRetries <= max {
+		return nil
+	}
+	if err == nil || err == errors.ErrBusySyncthing {
+		return errors.ErrUnknownSyncError
+	}
+	return err
+}
+
+// GetSyncthingStatus returns the syncthing status
+func (s *Syncthing) GetSyncthingStatus(ctx context.Context, folder *Folder, local bool) (string, error) {
+	params := map[string]string{
+		"folder":  GetFolderName(folder),
+		"since":   "0",
+		"limit":   "1",
+		"timeout": "0",
+		"events":  "StateChanged",
+	}
+	scList := []StateChangedEvent{}
+	body, err := s.APICall(ctx, "rest/events", "GET", 200, params, local, nil, true, 3)
+	if err != nil {
+		log.Infof("error getting events: %s", err.Error())
+		if strings.Contains(err.Error(), "Client.Timeout") {
+			return "", errors.ErrBusySyncthing
 		}
-		return err
+		return "", errors.ErrLostSyncthing
 	}
-	return nil
+
+	err = json.Unmarshal(body, &scList)
+	if err != nil {
+		log.Infof("error unmarshalling events: %s", err.Error())
+		return "", errors.ErrLostSyncthing
+	}
+
+	if len(scList) == 0 {
+		return "scanning", nil
+	}
+
+	return scList[0].Data.State, nil
 }
 
-// GetStatus returns the syncthing status
-func (s *Syncthing) GetStatus(ctx context.Context, folder *Folder, local bool) (*Status, error) {
-	params := getFolderParameter(folder)
-	status := &Status{}
-	body, err := s.APICall(ctx, "rest/db/status", "GET", 200, params, local, nil, true, 3)
+// GetPullErrors returns the syncthing current pull errors
+func (s *Syncthing) GetPullErrors(ctx context.Context, local bool) (int64, error) {
+	params := map[string]string{
+		"since":   "0",
+		"limit":   "1",
+		"timeout": "0",
+		"events":  "FolderSummary",
+	}
+	fsList := []FolderSummaryEvent{}
+	body, err := s.APICall(ctx, "rest/events", "GET", 200, params, local, nil, true, 3)
 	if err != nil {
-		log.Infof("error getting status: %s", err.Error())
+		log.Infof("error getting events: %s", err.Error())
 		if strings.Contains(err.Error(), "Client.Timeout") {
-			return nil, errors.ErrBusySyncthing
+			return 0, errors.ErrBusySyncthing
 		}
-		return nil, errors.ErrLostSyncthing
+		return 0, errors.ErrLostSyncthing
 	}
-	err = json.Unmarshal(body, status)
+
+	err = json.Unmarshal(body, &fsList)
 	if err != nil {
-		log.Infof("error unmarshalling status: %s", err.Error())
-		return nil, errors.ErrLostSyncthing
+		log.Infof("error unmarshalling events: %s", err.Error())
+		return 0, errors.ErrLostSyncthing
 	}
 
-	return status, nil
+	if len(fsList) == 0 {
+		return 0, nil
+	}
+
+	return fsList[0].Data.Summary.PullErrors, nil
 }
 
 // GetFolderErrors returns the last folder errors
-func (s *Syncthing) GetFolderErrors(ctx context.Context, folder *Folder, local bool) error {
-	params := getFolderParameter(folder)
-	params["since"] = "0"
-	params["limit"] = "1"
-	params["timeout"] = "0"
-	params["events"] = "FolderErrors"
-	folderErrorsList := []FolderErrors{}
+func (s *Syncthing) GetFolderErrors(ctx context.Context, local bool) error {
+	params := map[string]string{
```

Comment:
Is it not needed anymore the folder parameter?

Reply:
With the events API we can query errors on any folder in a single query